src/NLog/Config/LoggingConfigurationFileLoader.cs

```diff
@@ -248,6 +248,28 @@ public IEnumerable<string> GetDefaultCandidateConfigFilePaths(string fileName)
                         yield return Path.ChangeExtension(configurationFile.Replace(vshostSubStr, "."), ".nlog");
                     }
                 }
+#if NETSTANDARD && !NETSTANDARD1_3
+                else
+                {
+                    string processFilePath = ProcessIDHelper.Instance.CurrentProcessFilePath;
+                    string processDirectory = PathHelpers.TrimDirectorySeparators(Path.GetDirectoryName(processFilePath));
+                    if (!string.IsNullOrEmpty(entryAssemblyLocation) && !string.Equals(entryAssemblyLocation, processDirectory, StringComparison.OrdinalIgnoreCase))
+                    {
+                        // Handle dotnet-process loading .NET Core-assembly, or IIS-process loading website
+                        string assemblyFileName = Path.GetFileName(System.Reflection.Assembly.GetEntryAssembly().Location);
+                        if (!string.IsNullOrEmpty(assemblyFileName))
+                            yield return Path.Combine(entryAssemblyLocation, assemblyFileName + ".nlog");
+                        // Handle unpublished .NET Core Application
+                        assemblyFileName = Path.GetFileNameWithoutExtension(assemblyFileName);
+                        if (!string.IsNullOrEmpty(assemblyFileName))
+                            yield return Path.Combine(entryAssemblyLocation, assemblyFileName + ".exe.nlog");
+                    }
+                    else if (!string.IsNullOrEmpty(processFilePath))
+                    {
+                        yield return processFilePath + ".nlog";
+                    }
```

Comment:
Could you please elaborate the cases that here are implemented? Don't we need an unit test for this?

Reply:
Thought it was obvious as the other case is for running inside "dotnet.exe" then this case is for NOT running inside "dotnet.exe".

Reply:
Not that skilled with this build dynamic assemblies/processes and then running them. Find them very ugly and they only seem to be available for NetFramework (Not NetCore).

Reply:
> Could you please elaborate the cases that here are implemented? Don't we need an unit test for this?

I meant all the 3? (not counted) test cases

Reply:
> Not that skilled with this build dynamic assemblies/processes and then running them. Find them very ugly and they only seem to be available for NetFramework (Not NetCore).

I think we need to mock them, but I forgot that the code in NLog is not very mockable :(

Reply:
Will try and create new internal interface that merges ProcessIDHelper and IAppDomain. But not easy since IAppDomain and AppDomainWrapper are public. Then it should become easier to mock and exercise the config-lookup.

Reply:
> Will try and create new internal interface that merges ProcessIDHelper and IAppDomain

Don't think that is needed. We need to inject them both, or a factory 

I can give it a try if you like

Reply:
Well to me IAppDomain is the hosting environment-interface. Don't need a new interface for every property in the hosting environment.

Reply:
> I can give it a try if you like

Would rather prefer you review and merge: https://github.com/NLog/NLog.Extensions.Logging/pull/263 :)

Reply:
> Would rather prefer you review and merge: [NLog/NLog.Extensions.Logging#263](https://github.com/NLog/NLog.Extensions.Logging/pull/263) :)

done

So now: https://github.com/snakefoot/NLog/pull/7 :) (for the general idea)

Reply:
Already working on my own implementation of this mock. Your time would be better spent on all the other spider-issues in the hack-n-slash dungeon called NLog :)

Reply:
What a spider issue? ❓

Reply:
Guess you have never played Diablo or other hack-n-slash dungeon-games. They usually have spiders. And spiders are good at making cobwebs. For NLog it means all the issues detected by the code-analyzers you have been so nice to enable.